agent/util-scripts/pbench-tool-meister-start

```diff
@@ -244,13 +226,535 @@ def waitpid(pid):
     return exit_status
 
 
-def main(_prog, cli_params):
-    """Main program for the tool meister start.
+class StartTmsErr(ReturnCode.Err):
+    """StartTmsErr - derived from ReturnCode.Err, specifically raised by the
+    start_tms_via_ssh() method.
+    """
+
+    pass
+
+
+def start_tms_via_ssh(
+    exec_dir: Path,
+    ssh_cmd: str,
+    ssh_path: Path,
+    tool_group: str,
+    ssh_opts: str,
+    full_hostname: str,
+    redis_server,
+    redis_client: redis.Redis,
+    logger: logging.Logger,
+):
+    """start_tms_via_ssh - orchestrate the creation of local and remote Tool
+    Meister instances using ssh for those that are remote.
 
-    :cli_params: expects a CLI parameters object which has two attributes:
+    Raises a StartTmsErr on failure.
+
+    NOTE: all local and remote Tool Meisters are started even if failures
+    occur for some; this allows the user to see logs for all the individual
+    failures.
+    """
+    assert len(tool_group.hostnames) > 0, "Logic bomb!  No hosts to run tools"
+    failures = 0
+    successes = 0
+    tool_meister_cmd = exec_dir / "tool-meister" / "pbench-tool-meister"
+    base_args = [ssh_cmd]
+    base_args.extend(shlex.split(ssh_opts))
+    args = [
+        "<host replace me>",
+        f"{tool_meister_cmd}-remote",
+        redis_server.host,
+        str(redis_server.port),
+        "<tm param key>",
+        "yes",  # Yes, request the tool meister daemonize itself
+    ]
+    tms = dict()
+    tm_count = 0
+    for host in tool_group.hostnames.keys():
+        tm_count += 1
+        tm_param_key = f"tm-{tool_group.group}-{host}"
+        if host == full_hostname:
+            logger.debug("6a. starting localhost tool meister")
```

Comment:
Why "6a" instead of just "6" like line 1035?

Reply:
It was `5a` before ...